File: lanouOC/ClassDemo02/ClassDemo02/Student.m
//
//  Student.m
//  ClassDemo02
//
//  Created by Duger on 13-9-3.
//  Copyright (c) 2013年 Duger. All rights reserved.
//

#import "Student.h"


@implementation Student
@synthesize school,record;

+(id)studentWithName:(NSString *)_name identify:(NSString *)_identify sex:(NSString *)_sex age:(NSInteger)_age School: (NSString *)_school record:(NSString *)_record
{
    Student *student = [[Student alloc]initWithName:_name identify:_identify sex:_sex age:_age School:_school record:_record];
    return [student autorelease];
}

-(id)initWithName:(NSString *)_name identify:(NSString *)_identify sex:(NSString *)_sex age:(NSInteger)_age School: (NSString *)_school record:(NSString *)_record;
{
    
    if (self = [super initWithName:_name identify:_identify sex:_sex age:_age]) {
//        name = _name;
//        identify = _identify;
//        sex = _sex;
//        age = _age;
        school = _school;
        record = _record;
    }
    
    return self;
}


@end

File: C/StudentManagement/StudentManagement/student.c
//
//  student.c
//  StudentManagement
//
//  Created by Duger on 13-8-22.
//  Copyright (c) 2013年 Duger. All rights reserved.
//

#include <stdio.h>
#include <string.h>
#include "student.h"

Student maxOne(Student St[]){
    
    Student ageMin = {};
    Student scoreMax = {};
    ageMin = St[0];
    
    for (int i = 0; i < 3; i++) {
        
        if (St[i].score > scoreMax.score) {
            scoreMax = St[i];
        }
    }
    return  scoreMax;
    
    for (int i = 0; i < 3; i++) {
        
        if (St[i].age < ageMin.age) {
            ageMin = St[i];
        }
    }
    return ageMin;
}

void printStudent(Student stu){
    
    printf("%s分数最高,",stu.name);
    printf("分数为%.2f ",stu.score);
    printf("年龄为%d \n",stu.age);
    
}

void arrangement(Student St[]){
    Student temp = {};
    for (int i = 0; i < 5; i++) {
        for (int j = 0; j < 5 - i; j++) {
            if (St[j].score < St[j + 1].score ) {
                temp = St[j + 1];
                St[j + 1] = St[j];
                St[j] = temp;
            }
        }
    }

    
    for (int i = 0; i < 5; i++) {
        
        printf("%s,",St[i].name);
        printf("分数为%.2f ",St[i].score);
        printf("年龄为%d \n",St[i].age);
        
    }
//
//    temp2 = {St}
//    return temp2;
    
}

void printStudent2(Student St[]){
    
    for (int i = 0; i < 5; i++) {
        
    
    printf("%s分数最高,",St[i].name);
    printf("分数为%.2f ",St[i].score);
    printf("年龄为%d \n",St[i].age);
        
    }
    
}



